plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/utils/ToolkitUtils.kt

```diff
@@ -41,3 +46,32 @@ fun inspectExistingConnection(project: Project): Boolean =
 
         return@let false
     }
+
+
+fun Project.reloadToolkitToolWindow(isToolkitConnected: Boolean?) {
+    val isConnected = isToolkitConnected ?: inspectExistingConnection(this)
+    this.toggleToolkitToolWindow(isBrowser = !isConnected)
+}
+
+fun Project.toggleToolkitToolWindow(isBrowser: Boolean) {
+    val component = if (isBrowser) {
+        ToolkitWebviewPanel.getInstance(this).let {
+            it.browser?.prepareBrowser(BrowserState(FeatureId.AwsExplorer))
+            it.component
+        }
+    } else {
+        AwsToolkitExplorerToolWindow.getInstance(this)
+    }
+
+    val contentManager = AwsToolkitExplorerToolWindow.toolWindow(this).contentManager
+
+    val myContent = contentManager.factory.createContent(component, null, false).also {
+        it.isCloseable = true
+        it.isPinnable = true
+    }
+
+    runInEdt {
+        contentManager.removeAllContents(true)
```

Comment:
do we need to dispose the content or does that happen automatically

Reply:
yea I feel we should do this

Reply:
apparently I have not done this part yet (the entire browserPanel is not dispoable yet)

Reply:
lol

```
java.lang.RuntimeException: Memory leak detected: 'Module: 'Calculator'' (class com.intellij.workspaceModel.ide.impl.legacyBridge.module.ModuleBridgeImpl) was registered in Disposer as a child of 'ROOT_DISPOSABLE' (class com.intellij.openapi.util.Disposer$2) but wasn't disposed.
Register it with a proper parentDisposable or ensure that it's always disposed by direct Disposer.dispose call.
See https://jetbrains.org/intellij/sdk/docs/basics/disposers.html for more details.
The corresponding Disposer.register() stacktrace is shown as the cause:
```